### PR TITLE
urllib python 3 support and checking for windows platform

### PR DESCRIPTION
**Pull request by Bobsans**
nice tool =)

### CODE REVIEW

File: iconsextract.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 import tempfile
 import zipfile
-import urllib
 from xml.dom.minidom import parse
 
+try:
+    from urllib.request import urlretrieve
+except ImportError:
+    from urllib import urlretrieve
+
```
Comment (pereorga) [inline iconsextract.py:13]: Tested this and works great, thanks!

File: iconsextract.py [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 # TODO:
-#   windows only
-#   Python 3 support
+#   windows only                            --- implemented
+#   Python 3 support                        --- implemented
```
Comment (pereorga) [inline iconsextract.py]: We can get rid of these lines as they are no longer TODO items

File: iconsextract.py [hunk 3]
```diff
@@ -86,6 +91,10 @@ def download_iconsext():
 
 
 def main():
+    if platform.system().lower() != 'windows':
+        sys.stderr.write('This programm work only on Windows')
```
Comment (pereorga) [inline iconsextract.py]: Typo in "program". I would just write something like: "Windows is required."

File: iconsextract.py [hunk 4]
```diff
@@ -176,4 +185,4 @@ def main():
     shutil.rmtree(get_temp_directory())
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (pereorga) [inline iconsextract.py]: I would prefer to keep the line ending character

File: iconsextract.py (hunk 3 above)
Comment (Bobsans) [inline iconsextract.py]: yes, thanks. i bad know english.)

**Comment by pereorga** [comment]
Thanks @Bobsans

Can you submit a new PR with these few changes?

File: iconsextract.py (hunk 2 above)
Comment (Bobsans) [inline iconsextract.py]: yes. great idea)

File: iconsextract.py (hunk 4 above)
Comment (Bobsans) [inline iconsextract.py]: ok.

File: iconsextract.py [hunk 5]
```diff
@@ -69,24 +69,22 @@ def create_temp_directory():
...
-        if name.lower() == 'filetypesman.exe':
-            zip_file.extract(name, path)
-            break
-    zip_file.close()
+    with zipfile.ZipFile(os.path.join(path, 'filetypesman.zip')) as zip_file:
+        for name in zip_file.namelist():
+            if name.lower() == 'filetypesman.exe':
+                zip_file.extract(name, path)
+                break
     os.remove(os.path.join(path, 'filetypesman.zip'))
```
Comment (pereorga) [inline iconsextract.py:75]: Thanks

File: iconsextract.py [hunk 6]
```diff
@@ -69,24 +69,22 @@ def create_temp_directory():
...
-        if name.lower() == 'iconsext.exe':
-            zip_file.extract(name, path)
-            break
-    zip_file.close()
+    with zipfile.ZipFile(os.path.join(path, 'iconsext.zip')) as zip_file:
+        for name in zip_file.namelist():
+            if name.lower() == 'iconsext.exe':
+                zip_file.extract(name, path)
+                break
     os.remove(os.path.join(path, 'iconsext.zip'))
```
Comment (pereorga) [inline iconsextract.py:86]: Ok thanks

**Comment by Bobsans** [comment]
I do not understand what you mean speaking about changes and PR?)

**Comment by pereorga** [comment]
I meant a clean pull request, so less commits end up in the repo. But that's fine.

Applied, thanks

**Comment by Bobsans** [comment]
ок. i have a habit of all is kept constant.

**Pull request merged**